tfs/src/main/java/hudson/plugins/tfs/model/AbstractHookEvent.java

```diff
@@ -275,15 +277,32 @@ private Boolean branchMatches(final GitCodePushedEventArgs gitCodePushedEventArg
                     continue;
                 }
 
+                this.envVars = project instanceof Job ? buildEnv((Job) project) : new EnvVars();
+                List<Action> actionsWithSafeParams = actions;
+                if (project instanceof Job) {
+                    // Add job-specific additional varaiables to current run.
+                    final ArrayList<ParameterValue> values = getDefaultParameters((Job) project);
+                    final String vstsRefspec = getVstsRefspec(gitCodePushedEventArgs);
+                    values.add(new StringParameterValue("vstsRefspec", vstsRefspec));
+                    values.add(new StringParameterValue("vstsBranchOrCommit", gitCodePushedEventArgs.commit));
+                    SafeParametersAction paraAction = new SafeParametersAction(values);

```

Comment:
final?

Reply:
Good catch. Interesting CHECKSTYLE didn't report in this case. Added final to it.